book/src/tutorial/parser.md

```diff
@@ -56,13 +56,3 @@ The **second** parameter to a tracked function is **always** some kind of Salsa
 Tracked functions may take other arguments as well, though our examples here do not.
 Functions that take additional arguments are less efficient and flexible.
 It's generally better to structure tracked functions as functions of a single Salsa struct if possible.
-
-### The `returns(ref)` annotation
-
-You may have noticed that `parse_statements` is tagged with `#[salsa::tracked(returns(ref))]`.
-Ordinarily, when you call a tracked function, the result you get back is cloned out of the database.
-The `returns(ref)` attribute means that a reference into the database is returned instead.
-So, when called, `parse_statements` will return an `&Vec<Statement>` rather than cloning the `Vec`.
-This is useful as a performance optimization.
-(You may recall the `returns(ref)` annotation from the [ir](./ir.md) section of the tutorial,
-where it was placed on struct fields, with roughly the same meaning.)
```

Comment:
It would be nice if we move this block to the `ir` section and explain the `returns(ref)` (and deref etc) there

Reply:
I think it's better to explain it in the `parser.md` file because `ir` only talks about structs (input, tracked, interned) whereas `parser` section talks about tracked functions, I think adding `returns(ref)` there makes more sense.

Reply:
Yeah that could make sense. I added an explanation of `returns` for both functions and structs in the `parser` section, it could be split into `returns` for structs in `ir` and then `returns` for functions in `parser`.